File: thing.pl
# perl thing.pl
# author: slab
while (1) {
	my $input = <STDIN>;
	chomp $input;
	my @words = split(' ', $input);

	if (scalar @words < 2) {
		print "Invalid Input\n";
		next;
	}

	my $op = $words[0];
	my $result = $words[1];

	foreach (@words[2..$#words]) {
		if ($op eq "+") {
			$result += $_;
		} 
		elsif ($op eq "-") {
			$result -= $_;
		} 
		elsif ($op eq "*") {
			$result *= $_;
		} 
		elsif ($op eq "/") {
			$result /= $_;
		} 
		else {
			print "Invalid Operator\n";
			return;
		}
	}

	print "$result\n";
}

File: thing.rs
// rustc thing.rs && ./thing
// author: slab
fn main() {
	std::io::stdin().lines()
		.filter_map(|line| line.ok())
		.for_each(|line| {
			let mut words = line.split(' ');

			let sign = words.next();
			let words = words.filter_map(|word| word.parse::<i32>().ok());
			println!("{}", match sign {
				Some("+") => words.reduce(|a, b| a + b),
				Some("-") => words.reduce(|a, b| a - b),
				Some("*") => words.reduce(|a, b| a * b),
				Some("/") => words.reduce(|a, b| a / b),
				Some(_)   => panic!("Invalid operator"),
				None      => panic!("Invalid input")
			}.unwrap_or(0))
		});
}
